backend/src/nodes/impl/onnx/model.py

```diff
@@ -1,5 +1,69 @@
 # This class defines an interface.
 # It is important that is does not contain types that depend on ONNX.
-class OnnxModel:
+from typing import Union
+
+import re2
+
+re2_options = re2.Options()
+re2_options.dot_nl = True
+re2_options.encoding = re2.Options.Encoding.LATIN1
+
+U2NET_STANDARD = re2.compile(b"1959.+1960.+1961.+1962.+1963.+1964.+1965", re2_options)
+U2NET_CLOTH = re2.compile(
+    b"output.+d1.+Concat_1876.+Concat_1896.+Concat_1916.+Concat_1936.+Concat_1956",
+    re2_options,
+)
+U2NET_SILUETA = re2.compile(b"1808.+1827.+1828.+2296.+1831.+1850.+1958", re2_options)
+
+
+class OnnxGeneric:
     def __init__(self, model_as_bytes: bytes):
         self.bytes: bytes = model_as_bytes
+        self.arch = "generic"
+        self.sub_type = "Generic"
+
+
+class OnnxU2Net:
+    def __init__(self, model_as_bytes: bytes):
+        self.bytes: bytes = model_as_bytes
+        self.arch = "u2net"
+        self.sub_type = "RemBg"
+
+
+class OnnxU2NetCloth:
+    def __init__(self, model_as_bytes: bytes):
+        self.bytes: bytes = model_as_bytes
+        self.arch = "u2net_cloth"
+        self.sub_type = "RemBg"
```

Comment:
What's the motivation for making those 2 separate classes? Couldn't this just be one `OnnxRemBg` class?

Reply:
Kind of similar to why we have different classes for different PyTorch arches. I split them because I needed to distinguish them for output typing, but they are also distinct in that they require entirely different session wrappers.

Reply:
> I split them because I needed to distinguish them for output typing, but they are also distinct in that they require entirely different session wrappers

These are 2 classes are only used here. Everything else only uses `OnnxRemBgModel`, `OnnxGeneric`, and `OnnxModel`.

So what's the point of having 2 classes for rembg models?

Reply:
Because I needed to distinguish the u2net arch from the u2net_cloth arch, because the removeBackground type function uses those to type the output.